src/main/java/org/broadinstitute/consent/http/resources/DatasetResource.java

```diff
@@ -210,6 +209,14 @@ public Response patchByDatasetUpdate(@Auth AuthUser authUser,
         throw new BadRequestException("Properties are invalid");
       }
       Dataset patched = datasetRegistrationService.patchDataset(datasetId, user, patch);
+      if (patched != null && patched.getIndexedDate() != null) {
+        // If the dataset was indexed, we need to re-index it
+        try (var indexResponse = elasticSearchService.indexDataset(patched, user)) {
+          if (!HttpStatusCodes.isSuccess(indexResponse.getStatus())) {
```

Comment:
When this is called from `DatasetRegistrationService.updateDataset()`, the check is
```
        if (response.getStatus() >= 400) {
```

And in `DacResource.approveDataset()` the check is:
```
        if (indexResponse.getStatus() >= Status.BAD_REQUEST.getStatusCode()) {
```

I also see the same check done in `DatasetService.synchronizeDatasetInESIndex()`. Can all these cases be refactored into one method in `ElasticSearchService`?

Reply:
If so, I think this is the clearest pattern of those three.

Reply:
Yes, I think I can do some re-organizing. There are two distinct cases, one where we want to ensure that we are always performing both operations (update index and indexed date), and one where we only want to do it when it has been previously indexed.